### PR TITLE
Migrate to ES2015

### PR DESCRIPTION
**Pull request by devoto13**
Updated more files to ES2015 syntax.

I've added comments on interesting places, because GitHub produced quite messed up diff for this PR.

### CODE REVIEW

File: lib/executor.js [hunk 1]
```diff
@@ -1,60 +1,70 @@
...
 
-    if (!capturedBrowsers.length) {
-      log.warn('No captured browser, open %s//%s:%s%s', config.protocol, config.hostname,
-        config.port, config.urlRoot)
+    this.executionScheduled = false
+    this.pendingCount = 0
+    this.runningBrowsers = null
+
+    // bind all the events
+    this.emitter.on('run_complete', () => this.onRunComplete())
```
Comment (devoto13) [inline lib/executor.js]: Use explicit event binding as in the previous PR. The end goal would be to remove `bindAllEvents` completely and use explicit subscriptions.

File: lib/file.js [hunk 2]
```diff
@@ -1,35 +1,30 @@
-// File
-// ====
-//
-// File object used for tracking files in `file-list.js`
+'use strict'
 
-// Dependencies
-// ------------
+/**
```
Comment (devoto13) [inline lib/file.js]: I think it is good idea to convert comments to JSDoc, so they are picked up by IDE and show info on methods/classes. Do you think it is something worths doing?

File: lib/file-list.js [hunk 3]
```diff
@@ -40,214 +36,206 @@ function byPath (a, b) {
...
+  // Do the actual work of refreshing
+  _refresh () {
+    const buckets = this.buckets
+    const matchedFiles = new Set()
 
-    if (_.isEmpty(files)) {
-      log.warn('Pattern "%s" does not match any file.', pattern)
-      return
-    }
+    let promise
```
Comment (devoto13) [inline lib/file-list.js]: It seems that let/const use a bit different initialization logic, than var. I had to split this into declaration and initialization for tests to pass. They override scheduler for `Promise` to invoke methods synchronously and it throws `promise is not defined` after conversion.

File: lib/index.js [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
-var launcher = require('./launcher')
-var cfg = require('./config')
+const constants = require('./constants')
+const Server = require('./server')
+const runner = require('./runner')
+const stopper = require('./stopper')
+const launcher = require('./launcher')
+const cfg = require('./config')
 
 // TODO: remove in 1.0
```
Comment (devoto13) [inline lib/index.js]: It's 2.0 already :)
Shall we remove this method?

File: lib/init.js [hunk 5]
```diff
@@ -65,27 +69,28 @@ var installPackage = function (pkgName) {
...
 }
 
-var validateRequireJs = function (useRequire) {
+function validateRequireJs (useRequire) {
   if (useRequire) {
     validateFramework('requirejs')
   }
 }
 
+// TODO: update tests to not rely on internal module state and change this to const
```
Comment (devoto13) [inline lib/init.js]: The mocking library used in tests does not expose block-scoped declarations, so I had to keep `var` here. What do you think about exposing some of the methods from this file on `module.exports` to get rid special import in tests completely?

**Comment by devoto13** [comment]
Travis failed with a strange error message when validating commit message:

```
Retrieving relevant commit message for 65b5c4279f29dd38104178e245e259fdbd70119d
fatal: bad object 65b5c4279f29dd38104178e245e259fdbd70119d
```

File: lib/file.js (hunk 2 above)
Comment (johnjbarton) [reply lib/file.js]: Ok by me, but in a separate PR if you want to do it.

File: lib/index.js (hunk 4 above)
Comment (johnjbarton) [reply lib/index.js]: yes again in a separate PR.

File: lib/helper.js [hunk 6]
```diff
@@ -38,26 +41,26 @@ var parser = function (pattern, out) {
   return parser(pattern.substring(matches[0].length), out)
 }
 
-var gsParser = function (pattern, out) {
+function gsParser (pattern, out) {
   if (pattern === '**') {
```
Comment (johnjbarton) [inline lib/helper.js]: How about using `const gsParser = (pattern, out) => {` here?

Comment (johnjbarton) [reply lib/helper.js]: (and consistently of course)

File: lib/init.js (hunk 5 above)
Comment (johnjbarton) [reply lib/init.js]: Yes, that is better....again a separate PR ;-)

File: lib/helper.js (hunk 6 above)
Comment (devoto13) [reply lib/helper.js]: Fixed. Sorry, for the delay.

**Comment by johnjbarton** [comment]
I would like to see these changes land, but unfortunately I am not able to get a long block of time to review large changes.   If you send each of these as a PR and set me as reviewer I will try to get to them the same day.

**Comment by devoto13** [comment]
@johnjbarton Sure, will do it.

**Pull request closed without merging**